test/test_fat.rb

```diff
@@ -5,6 +5,40 @@
 class FatFileTest < Minitest::Test
   include Helpers
 
+  def test_empty_file
+    empty_file = tempfile_with_data("empty_file", "")
+
+    assert_raises MachO::NotAMachOError do
+      MachO::FatFile.new(empty_file)
+    end
+  ensure
+    empty_file.close
+    empty_file.unlink
```

Comment:
I think it would be good to get rid of this code duplication in all the `ensure`s. An untested idea would be to modify `tempfile_with_data` to remember the generated file (in an instance variable) and then clean it up in the `teardown` method (not sure how to put that into `Helpers` correctly so that test classes can still add their own `teardown` methods).